enhancements/cloud-integration/ibm/service-endpoint-dynamic-override.md

```diff
@@ -0,0 +1,126 @@
+
+## Motivation
+
+IBM Cloud requires this enhancement for their control plane replatforming efforts. OpenShift IPI for IBM Cloud will be used to deploy a cluster with critical responsibility in our managed control plane. During the genesis phase of region bringup, existing service endpoints will be used while new regional service endpoints are brought up. Once the new regional service endpoints are available, the aforementioned cluster's service endpoint configuration will need to be updated.
+
+### User Stories
+
+* As an OpenShift cluster administrator, I want to update my cluster's current service endpoint configuration to point to new service endpoints so that I can comply with administrative requirements to directs component traffic through regional or private endpoints.
+ 
+### Goals
+
+* Provide an official path for IBM deployed clusters to update the infrastructure object with service endpoint override(s) that will propagate to all dependent components without further user intervention.
+
+
+
+## Proposal
+
+To realize this enhancement:
+
+* Expand API definition to support defining services + endpoints within cloud provider spec for IBM
+* Modify CCCMO so that changes are reconciled from infrastructure spec to status for IBM cloud provider and cloud config
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible managing an existing openshift custer deployed on IBM infrastructure.
+
+1. The cluster administrator wishes to use private IBM Cloud endpoints.
+2. The cluster administrator identifies the services that they wish to update (ie IAM and resource controller) and identifies the endpoints for these services.
+3. The cluster administrator updates the infrastructure object to contain a list of overrides where each element is the name of the service and the endpoint to use for that service. `oc edit infrastructure  -n default cluster`
+4. Once the service endpoint override update has been processed/reconciled, components can act on the change (if applicable) and use in future operations (note: may need to be restarted to pick up the change).
+
+**cccmo** is an operator responsible for watching updates to the infrastructure object and perforning updates once any value(s) are set. 
+
+1. The cccmo reconciliation loop observes that the IBMCloudPlatform spec within the infrastructure object has been set,
+2. The cccmo validates the endpoints and then updates the IBMCloudPlatformStatus and cloud config.
+
+
+### API Extensions
+
+* Extend `IBMCloudPlatformSpec` to contain service endpoint field that users may define as desired overrides.
+
+```
+type IBMCloudPlatformSpec struct {
+	// serviceEndpoints is a list of custom endpoints which will override the default
+	// service endpoints of an IBM Cloud service. These endpoints are consumed by
+	// components within the cluster to reach the respective IBM Cloud Services.
```

Comment:
We need to explain what happens here when there is a change to this, how are values reconciled with status etc